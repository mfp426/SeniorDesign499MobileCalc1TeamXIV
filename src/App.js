import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom";
import './App.css';
import { useState } from 'react';
import NavBar from './components/Navbar.js';
import Calculator from "./pages/CalculatorPage";
import HomePage from "./pages/HomePage";
import ConversionPage from "./pages/CategoryPages/Conversion/ConversionPage";
import KinEnergyEqvSpeedLGPage from "./pages/CategoryPages/KinEnergyEqvSpeedLG/KinEnergyEqvSpeedLGPage";
import SpeedAndVelocityPage from "./pages/CategoryPages/Conversion/FormulaPage/SpeedAndVelocityPage";
import SpeedSlidingToStop from "./pages/CategoryPages/KinEnergyEqvSpeedLG/FormulaPage/SpeedSlidingToStopPage";
import Footer from "./components/Footer.js";
import VelocitySlidingToStop from "./pages/CategoryPages/KinEnergyEqvSpeedLG/FormulaPage/VelocitySlidingToStopPage";
import GradeAndSuperElevationPage from "./pages/CategoryPages/GradeAndSuperElevation/GradeAndSuperElevationPage";
import PercentGradePage from "./pages/CategoryPages/GradeAndSuperElevation/FormulaPage/PercentGradePage";
import PercentSuperelevationPage from "./pages/CategoryPages/GradeAndSuperElevation/FormulaPage/PercentSuperelevationPage";
import EDRPage from "./pages/CategoryPages/EDR/EDRPage";
import SpeedAtImpactPage from "./pages/CategoryPages/EDR/FormulaPage/SpeedAtImpactPage";
import ConstUniAvgEquationPage from "./pages/CategoryPages/ConstUniAvgEquation/ConstUniAvgEquationPage";
import ConstantDistancePage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantDistancePage";
import ConstantVelocityPage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantVelocityPage";
import ConstantTimePage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantTimePage";
import SlidetoStopDistWithDragPage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/SlidetoStopDistWithDragPage.js"
import SpeedAtImpactResultsPage from "./pages/CategoryPages/EDR/FormulaPage/SpeedAtImpactResultsPage";
import CenterOfMassEquationPage from "./pages/CategoryPages/CenterOfMassEquation/CenterOfMassEquationPage";
import COMLateralLeftPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLateralLeftPage";
import COMLateralRightPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLateralRightPage";
import COMLongitudinalFrontPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLongitudinalFrontPage";
import COMLongitudinalRearPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLongitudinalRearPage";

function App() {

    const [toggle, setToggle] = useState(true);

    const toggleState = () => setToggle(!toggle);




  return (
    <div className="mainContainer">

      <Router>
        <NavBar />
        <Routes>
          <Route path="/" element={<HomePage toggleState={toggleState}/>} />
          {/* Main Pages*/}
          <Route path = "/Calculator" element={<Calculator toggleState={toggleState}/>} />
          <Route path = "/HomePage" element={<HomePage toggleState={toggleState}/>} />
          {/* Conversion */}
          <Route path = "/Conversions" element={<ConversionPage toggleState={toggleState}/>} />
          <Route path = "/Conversions/SpeedAndVelocity" element={<SpeedAndVelocityPage toggleState={toggleState}/>} />
          {/*Kinematic Energy Equivalent */}
          <Route path = "/KinEnergyEqvSpeedLGPage" element={<KinEnergyEqvSpeedLGPage toggleState= {toggleState} />}/>
          <Route path = "/SpeedSlidingToStopPage" element = {<SpeedSlidingToStop toggleState={toggleState}/>}/>
          <Route path = "/VelocitySlidingtoStopPage" element = {<VelocitySlidingToStop toggleState = {toggleState}/>}/>
          {/* Grade and Super elevation */}
          <Route path = "/GradeAndSuperElevationPage" element={<GradeAndSuperElevationPage toggleState= {toggleState}/>}/>
          <Route path = "/PercentGradePage" element={<PercentGradePage toggleState={toggleState}/>}/>
          <Route path = "/PercentSuperelevationPage" element={<PercentSuperelevationPage toggleState={toggleState}/>}/>
          {/* EDR */}
          <Route path = "/EDR" element={<EDRPage toggleState={toggleState}/>}/>
          <Route path = "/EDR/SpeedAtImpact" element={<SpeedAtImpactPage/>}/>
          <Route path ="/EDR/SpeedAtImpact/Results" element={<SpeedAtImpactResultsPage/>} />
          {/* Constant Uniform Acceleration Equations */}
          <Route path = "/ConstUniAvgEquationPage" element={<ConstUniAvgEquationPage/>}/>
          <Route path = "/ConstantVelocityPage" element = {<ConstantVelocityPage/>}/>
          <Route path = "/ConstantTimePage" element = {<ConstantTimePage/>}/>
          <Route path = "/ConstantDistancePage" element={<ConstantDistancePage/>}/>
          <Route path = "/SlidetoStopDistWithDragPage" element ={<SlidetoStopDistWithDragPage/>}/>
          {/* Center of Mass Equations */}
          <Route path = "/CenterOfMassEquationPage" element ={<CenterOfMassEquationPage/>}/>
          <Route path = "/COMLateralLeftPage" element ={<COMLateralLeftPage/>}/>
          <Route path = "/COMLateralRightPage" element ={<COMLateralRightPage/>}/>
          <Route path = "/COMLongitudinalFrontPage" element ={<COMLongitudinalFrontPage/>}/>
          <Route path = "/COMLongitudinalRearPage" element ={<COMLongitudinalRearPage/>}/>
        </Routes>
        <Footer/>
      </Router>
    </div>
  );
}

export default App;
