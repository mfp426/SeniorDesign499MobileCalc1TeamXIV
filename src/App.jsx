import {BrowserRouter as Router, Route, Routes,} from "react-router-dom";
import {createContext, useEffect, useState} from 'react';

import './App.css';
import NavBar from './components/Navbar';
import Calculator from "./pages/CalculatorPage";
import HomePage from "./pages/HomePage";
import SearchPage from "./pages/SearchPage";
import ConversionPage from "./pages/CategoryPages/Conversion/ConversionPage";
import KinEnergyEqvSpeedLGPage from "./pages/CategoryPages/KinEnergyEqvSpeedLG/KinEnergyEqvSpeedLGPage";
import SpeedAndVelocityPage from "./pages/CategoryPages/Conversion/FormulaPage/SpeedAndVelocityPage";
import SpeedVelocitySlidingToStop from "./pages/CategoryPages/KinEnergyEqvSpeedLG/FormulaPage/SpeedVelocitySlidingToStopPage";
import GradeAndSuperElevationPage from "./pages/CategoryPages/GradeAndSuperElevation/GradeAndSuperElevationPage";
import PercentGradePage from "./pages/CategoryPages/GradeAndSuperElevation/FormulaPage/PercentGradePage";
import PercentSuperelevationPage from "./pages/CategoryPages/GradeAndSuperElevation/FormulaPage/PercentSuperelevationPage";
import EDRPage from "./pages/CategoryPages/EDR/EDRPage";
import SpeedAtImpactPage from "./pages/CategoryPages/EDR/FormulaPage/SpeedAtImpactPage";
import ConstUniAvgEquationPage from "./pages/CategoryPages/ConstUniAvgEquation/ConstUniAvgEquationPage";
import ConstantDistancePage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantDistancePage";
import ConstantVelocityPage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantVelocityPage";
import ConstantTimePage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/ConstantTimePage";
import SlidetoStopDistWithDragPage from "./pages/CategoryPages/ConstUniAvgEquation/FormulaPage/SlidetoStopDistWithDragPage"
import SpeedAtImpactResultsPage from "./pages/CategoryPages/EDR/FormulaPage/SpeedAtImpactResultsPage";
import CenterOfMassEquationPage from "./pages/CategoryPages/CenterOfMassEquation/CenterOfMassEquationPage";
import COMLateralPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLateralPage";
import COMLongitudinalPage from "./pages/CategoryPages/CenterOfMassEquation/FormulaPage/COMLongitudinalPage";
import RadiusCalculationPage from "./pages/CategoryPages/RadiusCalculation/RadiusCalculationPage";
import AcceDeceRatePage from "./pages/CategoryPages/AcceDeceRate/FormulaPage/AcceDeceRatePage.jsx";
import KinEnergyAndWorkPage from "./pages/CategoryPages/KinEnergyAndWork/KinEnergyAndWorkPage.jsx";
import KinEnergyCalPage from "./pages/CategoryPages/KinEnergyAndWork/KinEnergyCalPage.jsx";
import KinEnergyWithMassVelPage from "./pages/CategoryPages/KinEnergyAndWork/FormulaPage/KinEnergyWithMassVelPage.jsx";
import KinEnergyWithWeightSpdOrVelPage from "./pages/CategoryPages/KinEnergyAndWork/FormulaPage/KinEnergyWithWeightSpdOrVelPage.jsx";
import WorkCalPage from "./pages/CategoryPages/KinEnergyAndWork/WorkCalPage.jsx";
import WorkWithForceAndDistPage from "./pages/CategoryPages/KinEnergyAndWork/FormulaPage/WorkWithForceAndDistPage.jsx";
import WorkWithWeightAndDistPage from "./pages/CategoryPages/KinEnergyAndWork/FormulaPage/WorkWithWeightAndDistPage.jsx";
import MphAndFpsDueToKEPage from "./pages/CategoryPages/KinEnergyAndWork/FormulaPage/MphAndFpsDueToKEPage.jsx";

export const carContext = createContext({make: "", model: ""});


function App() {
    const [car, setCar] = useState({make: "", model: ""});

    useEffect(() => {
        console.log("Car: " + car.make + " " + car.model);
    }, [car]);


    return (
        <div>
            <carContext.Provider value={{car: car, setCar: setCar}}>
                <Router>
                    <NavBar />
                    <Routes>
                        <Route path="/" element={<Calculator/>} />

                        {/* Main Pages*/}
                        <Route path="/Calculator" element={<Calculator/>} />
                        <Route path="/HomePage" element={<HomePage/>} />
                        <Route path="/SearchPage" element={<SearchPage/>} />

                        {/* Conversion */}
                        <Route path="/Conversions" element={<ConversionPage/>} />
                        <Route path="/Conversions/SpeedAndVelocity" element={<SpeedAndVelocityPage/>} />

                        {/*Kinematic Energy Equivalent */}
                        <Route path="/KinEnergyEqvSpeedLGPage" element={<KinEnergyEqvSpeedLGPage/>} />
                        <Route path="/SpeedVelocitySlidingToStopPage" element={<SpeedVelocitySlidingToStop/>} />

                        {/* Grade and Super elevation */}
                        <Route path="/GradeAndSuperElevationPage" element={<GradeAndSuperElevationPage/>} />
                        <Route path="/PercentGradePage" element={<PercentGradePage/>} />
                        <Route path="/PercentSuperelevationPage" element={<PercentSuperelevationPage/>} />

                        {/* EDR */}
                        <Route path="/EDR" element={<EDRPage/>} />
                        <Route path="/EDR/SpeedAtImpact" element={<SpeedAtImpactPage />} />
                        <Route path="/EDR/SpeedAtImpact/Results" element={<SpeedAtImpactResultsPage />} />

                        {/* Constant Uniform Acceleration Equations */}
                        <Route path="/ConstUniAvgEquationPage" element={<ConstUniAvgEquationPage />} />
                        <Route path="/ConstantVelocityPage" element={<ConstantVelocityPage />} />
                        <Route path="/ConstantTimePage" element={<ConstantTimePage />} />
                        <Route path="/ConstantDistancePage" element={<ConstantDistancePage />} />
                        <Route path="/SlidetoStopDistWithDragPage" element={<SlidetoStopDistWithDragPage />} />

                        {/* Center of Mass Equations */}
                        <Route path="/CenterOfMassEquationPage" element={<CenterOfMassEquationPage />} />
                        <Route path="/COMLateralPage" element={<COMLateralPage />} />
                        <Route path="/COMLongitudinalPage" element={<COMLongitudinalPage />} />

                        {/*Radius Calculation*/}
                        <Route path="/RadiusCalculationPage" element={<RadiusCalculationPage />} />

                        <Route path={"/AcceDeceRatePage"} element={<AcceDeceRatePage />} />

                        {/*Kinetic Energy and Work*/}
                        <Route path={"/KinEnergyAndWorkPage"} element={<KinEnergyAndWorkPage />} />
                        <Route path={"/KinEnergyCalPage"} element={<KinEnergyCalPage />} />
                        <Route path={"/KinEnergyWithMassVelPage"} element={<KinEnergyWithMassVelPage />} />
                        <Route path={"/KinEnergyWithWeightSpdOrVelPage"} element={<KinEnergyWithWeightSpdOrVelPage />} />
                        <Route path={"/WorkCalPage"} element={<WorkCalPage />} />
                        <Route path={"/WorkWithWeightAndDistPage"} element={<WorkWithWeightAndDistPage />} />
                        <Route path={"/WorkWithForceAndDistPage"} element={<WorkWithForceAndDistPage />} />
                        <Route path={"/MphAndFpsDueToKEPage"} element={<MphAndFpsDueToKEPage />} /> 

                    </Routes>
                </Router>
            </carContext.Provider>
        </div>
    );
}

export default App;
