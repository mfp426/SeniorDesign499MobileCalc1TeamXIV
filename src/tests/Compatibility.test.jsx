import React from 'react';
import { render } from '@testing-library/react';
import App from '../App.jsx';

test('renders without crashing', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
   
test('renders correctly on web version', () => {
    // Set the window width to simulate a web view
    global.innerWidth = 1024;
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  
test('renders correctly on tablet version', () => {
    // Set the window width to simulate a tablet view
    global.innerWidth = 768;
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('renders correctly on mobile version', () => {
    // Set the window width to simulate a mobile view
    global.innerWidth = 375;
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });