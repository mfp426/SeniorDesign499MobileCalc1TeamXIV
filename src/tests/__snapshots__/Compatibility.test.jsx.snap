// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly on mobile version 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar  navbar-expand navbar-light bg-light "
    >
      <div
        class="container-fluid "
      >
        <ul
          class="nav navbar-nav "
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/SearchPage"
            >
              Search
            </a>
          </li>
          <li
            class="nav-item"
          >
            <div
              class="input-group"
            >
               
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="react-autowhatever-1"
                class="react-autosuggest__container"
                role="combobox"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-1"
                  autocomplete="off"
                  class="form-control"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <div
                  class="react-autosuggest__suggestions-container"
                  id="react-autowhatever-1"
                  role="listbox"
                />
              </div>
              <div
                class="input-group-append"
              />
            </div>
             
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container mb-5 center"
    >
      <div
        class="box justify-content-center"
      >
        <h2
          class="text-xl page-header mt-3 mb-3"
        >
          Categories
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-4 lg:grid-cols-4 gap-4"
        >
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Conversion
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy Equivalent Speed Loss/Gain
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Grade and Superelevation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            EDR
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Constant, Uniform, or Average Kinematic Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Center of Mass Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Radius Calculation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Acceleration / Deceleration rate when Acceleration / Deceleration factor is known
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy and Work
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly on tablet version 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar  navbar-expand navbar-light bg-light "
    >
      <div
        class="container-fluid "
      >
        <ul
          class="nav navbar-nav "
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/SearchPage"
            >
              Search
            </a>
          </li>
          <li
            class="nav-item"
          >
            <div
              class="input-group"
            >
               
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="react-autowhatever-1"
                class="react-autosuggest__container"
                role="combobox"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-1"
                  autocomplete="off"
                  class="form-control"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <div
                  class="react-autosuggest__suggestions-container"
                  id="react-autowhatever-1"
                  role="listbox"
                />
              </div>
              <div
                class="input-group-append"
              />
            </div>
             
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container mb-5 center"
    >
      <div
        class="box justify-content-center"
      >
        <h2
          class="text-xl page-header mt-3 mb-3"
        >
          Categories
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-4 lg:grid-cols-4 gap-4"
        >
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Conversion
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy Equivalent Speed Loss/Gain
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Grade and Superelevation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            EDR
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Constant, Uniform, or Average Kinematic Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Center of Mass Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Radius Calculation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Acceleration / Deceleration rate when Acceleration / Deceleration factor is known
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy and Work
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly on web version 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar  navbar-expand navbar-light bg-light "
    >
      <div
        class="container-fluid "
      >
        <ul
          class="nav navbar-nav "
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/SearchPage"
            >
              Search
            </a>
          </li>
          <li
            class="nav-item"
          >
            <div
              class="input-group"
            >
               
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="react-autowhatever-1"
                class="react-autosuggest__container"
                role="combobox"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-1"
                  autocomplete="off"
                  class="form-control"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <div
                  class="react-autosuggest__suggestions-container"
                  id="react-autowhatever-1"
                  role="listbox"
                />
              </div>
              <div
                class="input-group-append"
              />
            </div>
             
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container mb-5 center"
    >
      <div
        class="box justify-content-center"
      >
        <h2
          class="text-xl page-header mt-3 mb-3"
        >
          Categories
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-4 lg:grid-cols-4 gap-4"
        >
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Conversion
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy Equivalent Speed Loss/Gain
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Grade and Superelevation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            EDR
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Constant, Uniform, or Average Kinematic Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Center of Mass Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Radius Calculation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Acceleration / Deceleration rate when Acceleration / Deceleration factor is known
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy and Work
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar  navbar-expand navbar-light bg-light "
    >
      <div
        class="container-fluid "
      >
        <ul
          class="nav navbar-nav "
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/Calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/SearchPage"
            >
              Search
            </a>
          </li>
          <li
            class="nav-item"
          >
            <div
              class="input-group"
            >
               
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="react-autowhatever-1"
                class="react-autosuggest__container"
                role="combobox"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-1"
                  autocomplete="off"
                  class="form-control"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <div
                  class="react-autosuggest__suggestions-container"
                  id="react-autowhatever-1"
                  role="listbox"
                />
              </div>
              <div
                class="input-group-append"
              />
            </div>
             
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container mb-5 center"
    >
      <div
        class="box justify-content-center"
      >
        <h2
          class="text-xl page-header mt-3 mb-3"
        >
          Categories
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-4 lg:grid-cols-4 gap-4"
        >
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Conversion
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy Equivalent Speed Loss/Gain
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Grade and Superelevation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            EDR
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Constant, Uniform, or Average Kinematic Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Center of Mass Equations
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Radius Calculation
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Acceleration / Deceleration rate when Acceleration / Deceleration factor is known
          </button>
          <button
            class="btn btn-primary mt-4 text-truncate"
          >
            Kinetic Energy and Work
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
